File: Java/Cristian Maldonado/Matemática/Basic_Primality_Test.java
//Tener en cuenta que el algoritmo puede ser
//demasiado lento con una cantidad de consultas considerable
public class Primality_Test {

   static boolean isPrime(int x){
       if(x < 2){
           return false;
       }
       if(x == 2){
           return true;
       }
       if(x % 2 == 0){
           return false;
       }
       for (int i = 3; i * i <= x; i++) {
           if(x % i == 0){
               return false;
           }
       }
       return true;
   }
    
    public static void main(String[] args) {
        int n = 14;
        if(isPrime(n)){
            System.out.println("Es primo");
        }else{
            System.out.println("No es primo");
        }
    }   
}

File: Codigos estudiantes por lenguaje/C++/Cristian Maldonado/Matemática/Basic_Primality_Test.cpp
//Tener en cuenta que el algortimo puede ser
//demasiado lento con una cantidad de consultas considerables
#include <iostream>

using namespace std;

bool isPrime(int x){
    if(x < 2){
        return false;
    }
    if(x == 2){
        return true;
    }
    if(x % 2 == 0){
        return false;
    }
    for(int i = 3; i * i <= x; i++){
        if(x % i == 0){
            return false;
        }
    }
    return true;
}

int main()
{
    int n;
    scanf("%d", &n);
    if(isPrime(n)){
        printf("Es primo\n");
    }else{
        printf("No es primo\n");
    }
    return 0;
}
